### ISSUE TITLE
Android Performance Improvement

### ISSUE THREAD

**Issue by Sergeypt423** [react-community/lottie-react-native#333]
Hi
I've used your lottie-react-native package for a month and realised that it's performance is not good.
iOS version has no problem but android version - it's freezing and missing many frames.
Is that possible to improve performance by adding custom props to connect to the native lottie android?
Please let me know.
Thanks

**Comment by pedrosimao** [comment]
Hello, I am also working with Sergey on this project.
We are willing to work on new versions of Lottie react-native modules and contributing to this repo. We are just not sure what are the biggest performance bottlenecks and if they can be improved by adding props, for example. Also, do Lottie animation make full use of GPU?

**Comment by poberwong** [comment]
+1，It's missing many frames when in [react-navigaiton](https://github.com/react-navigation/react-navigation).

**Comment by pedrosimao** [comment]
@poberwong have you tested without the react-navigation? Does it change the fps?
Have you tried WIX react navigation?

**Comment by poberwong** [comment]
@pedrosimao sure, it runs more smoothly and has higher fps without  react-navigation

**Comment by pedrosimao** [comment]
@poberwong What is the exact difference in terms of fps with and without react-navigation. And have you tested react-native-navigation from Wix? (https://wix.github.io/react-native-navigation/ )

**Comment by Winglonelion** [comment]
hm, i usually control Lottie by a Animated Value and start it with `useNativeDriver`. In almost times, it worked fine.

**Comment by pedrosimao** [comment]
@Winglonelion can you share a code snippet? using "useNativeDriver" makes animation faster on Android?

**Comment by Winglonelion** [comment]
This is my `utils components to handle lottie. You can mix it with severals `Layout-support components` like [wix/react-native-interactable](https://github.com/wix/react-native-interactable)
```
import PropTypes from 'prop-types';
import React from 'react';
import {
  View,
  Easing,
  Animated,
  ViewPropTypes,
} from 'react-native';
import Lottie from 'lottie-react-native';

import styles, { initStyle } from './styles';

const loading = require('@images/lottie/trail_loading');

export default class LottieAnimation extends React.PureComponent {
  static propTypes = {
    icon: PropTypes.any,
    loop: PropTypes.bool,
    size: PropTypes.number,
    autoRun: PropTypes.bool,
    style: ViewPropTypes.style,
    progress: PropTypes.number,
    duration: PropTypes.number,
    resizeMode: PropTypes.string,
    iconStyle: ViewPropTypes.style,
    useNativeDriver: PropTypes.bool,
  }

  static defaultProps = {
    size: 100,
    loop: true,
    progress: 0,
    duration: 0,
    autoRun: true,
    icon: loading,
    iconStyle: null,
    style: undefined,
    resizeMode: 'cover',
    useNativeDriver: false,
  }

  state = { progress: new Animated.Value(this.props.progress || 0) };

  runIn = () => {
    const { duration, useNativeDriver } = this.props;
    Animated.timing(this.state.progress, {
      toValue: 1,
      easing: Easing.linear,
      duration,
      useNativeDriver,
    }).start();
  }

  runOut = () => {
    const { duration, useNativeDriver } = this.props;
    Animated.timing(this.state.progress, {
      toValue: 0,
      easing: Easing.linear,
      duration,
      useNativeDriver,
    }).start();
  }

  componentDidMount() {
    if (this.props.autoRun) {
      if (this.props.duration) this.runIn();
      else this.lottie.play();
    }
  }

  render() {
    const {
      loop,
      size,
      icon,
      iconStyle,
      resizeMode,
    } = this.props;

    const iStyle = initStyle(size);
    return (
      <View style={[ styles.default, this.props.style ]}>
        <View>
          <Lottie
            loop={loop}
            source={icon}
            resizeMode={resizeMode}
            ref={this.setLottieRef}
            progress={this.state.progress}
            style={[ styles.lottie, iStyle.icon, iconStyle ]}
          />
        </View>
      </View>
    );
  }

  setLottieRef = ref => { this.lottie = ref; }
}

```

**Comment by Winglonelion** [comment]
@pedrosimao  I think you must fork and override [lottie-android ](https://github.com/airbnb/lottie-android)project to achieving your goal.

But by what i knew, you shouldn't enable `hardware acceleration` in application level. Many other graphic libraries do not ship with `hardware acceleration`. And itself not well supporting for low android API.

Read more about hardware acceleration here: https://developer.android.com/guide/topics/graphics/hardware-accel

**Comment by pedrosimao** [comment]
I saw this on the Java code:
```
  @ReactProp(name = "hardwareAccelerationAndroid")
  public void setHardwareAcceleration(LottieAnimationView view, boolean use) {
    view.useHardwareAcceleration(use);
  }
```
And I guessed it was possible to use a `hardwareAccelerationAndroid` propo on my `LottieView`component. Unhappily it doesn't seem to work. It is a pitty that Lottie for React Native has so much less options than Android and iOS versions.

**Comment by Winglonelion** [comment]
it not be descible in Lottie document. Did you try it ? It maybe an Easter egg. 🥇

**Comment by pedrosimao** [comment]
@Winglonelion I have tried and it works! Documentation of React Native Lottie is not up to date.
I think for it work you also have to activate hardware acceleration on AndroidManifest.xml
There is only a catch. Some animations might look weird, especially on old devices. For simple animations, I think it works just great. You can try to see what you think.
Also all my problems came from opacity animations using Animated API, with hardware acceleration slugish performance is gone.

I am also thinking about implementing the fps props on react native Lottie. Android and iOS already have it. Do you want to help me with that? I have taken a look to see how they did with `hardwareAccelerationAndroid` and it seems quite trivial.

**Comment by defualt** [comment]
I had this problem and upgrade from 2.5.6 to 2.5.8 and it fixed it.

**Comment by emilioicai** [comment]
closing as this should be resolved on 2.5.8
